resources/terraform/aws-eks/main.tf

```diff
@@ -0,0 +1,185 @@
+resource "random_string" "name_suffix" {
+  length  = 5
+  lower   = true
+  upper   = false
+  numeric = true
+  special = false
+}
+
+locals {
+  name         = "testenv-eks-${random_string.name_suffix.result}"
+  cluster_name = "${local.name}-cluster"
+}
+
+provider "aws" {
+  region = var.region
+}
+
+provider "kubernetes" {
+  host                   = module.eks.cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
+
+  exec {
+    api_version = "client.authentication.k8s.io/v1"
+    command     = "aws"
+    # This requires the awscli to be installed locally where Terraform is executed
+    args = ["eks", "get-token", "--cluster-name", module.eks.cluster_id]
```

Comment:
You'll might need to leverage the `--region` flag with this command since there's no guarantee the CI machines will have that set and I'm unsure if the `aws` command will inherit the settings of `provider "aws"`, but I highly doubt it.